src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/ExternalSettingsProviderService.cs

```diff
@@ -0,0 +1,229 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Utilities.UnifiedSettings;
+using NuGet.Configuration;
+using NuGet.VisualStudio;
+
+namespace NuGet.PackageManagement.VisualStudio.Services
+{
+    [Guid("6C09BBE2-4537-48B4-87D8-01BF5EB75901")]
+    public sealed class ExternalSettingsProviderService : IExternalSettingsProvider
+    {
+        private const string MonikerAllowRestoreDownload = "packageRestore.allowRestoreDownload";
+        private const string MonikerPackageRestoreAutomatic = "packageRestore.packageRestoreAutomatic";
+        private const string MonikerSkipBindingRedirects = "bindingRedirects.skipBindingRedirects";
+        private const string MonikerDefaultPackageManagementFormat = "packageManagement.defaultPackageManagementFormat";
+        private const string MonikerPackageReference = "package-reference";
+        private const string MonikerPackagesConfig = "packages-config";
+        private const string MonikerShowPackageManagementChooser = "packageManagement.showPackageManagementChooser";
+
+        private readonly ISettings _settings;
+        private readonly VSSettings _vsSettings;
+
+        private PackageRestoreConsent _packageRestoreConsent;
+        private BindingRedirectBehavior _bindingRedirectBehavior;
+        private PackageManagementFormat _packageManagementFormat;
+
+        public ExternalSettingsProviderService()
+        {
+            var componentModel = NuGetUIThreadHelper.JoinableTaskFactory.Run(ServiceLocator.GetComponentModelAsync);
+            _settings = componentModel.GetService<ISettings>();
+            _vsSettings = _settings as VSSettings;
+            if (_vsSettings != null)
+            {
+                _vsSettings.SettingsChanged += VsSettings_SettingsChanged;
+            }
+            Debug.Assert(_settings != null);
+        }
+
+        private void VsSettings_SettingsChanged(object sender, EventArgs e)
+        {
+            _packageRestoreConsent = null;
+            _bindingRedirectBehavior = null;
+            _packageManagementFormat = null;
+            SettingValuesChanged.Invoke(this, ExternalSettingsChangedEventArgs.SomeOrAll);
```

Comment:
```suggestion
            SettingValuesChanged?.Invoke(this, ExternalSettingsChangedEventArgs.SomeOrAll);
```

Auto-events have a null value when nothing has subscribed to the event.

Reply:
I believe this is highly theoretical, as USX solely and (probably always?) subscribes to the event. Added the null-check anyway.